solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -586,6 +587,17 @@ private Request makeRequest(SolrRequest<?> solrRequest, String collection)
   }
 
   private void decorateRequest(Request req, SolrRequest<?> solrRequest) {
+    SolrRequest.SolrClientContext context = getContext();
+    req.header(CommonParams.SOLR_REQUEST_CONTEXT_PARAM, context.toString());
+    if (context == SolrRequest.SolrClientContext.CLIENT
+        || solrRequest.getParams().getBool(ShardParams.SHARDS_TOLERANT, false)) {
```

Comment:
Whole purpose of ratelimiter to limit the resources on all nodes. Not sure we really need to make any special case here.

Reply:
For our case in particular, practically we actually do want to avoid failing any requests that are `shards.tolerant=false`. Notably these are also the most likely requests to be retried on failure, so if we end up repeatedly executing requests on all nodes, only to repeatedly fail because of 1 struggling node (for example), that could easily cause load to increase on the other nodes to the point where the problem spreads to the entire cluster.